### PR TITLE
[FLINK-6586] InputGateMetrics return 0 as minimum for local channels

### PR DESCRIPTION
**Pull request by zentol**
Should be merged for 1.3 and 1.4.

### CODE REVIEW

**Comment by greghogan** [comment]
Does an `InputGate` contain both local and remote `InputChannel` such that the minimum positive value might be more desirable in that case?

**Comment by zentol** [comment]
That's a great point, haven't considered that. It may be easier to just compare the final result with `Integer.MAX_VALUE` and return `0` if the match.

**Comment by zentol** [comment]
Updated the PR.

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGateMetrics.java [hunk 1]
```diff
@@ -86,6 +86,9 @@ int refreshAndGetMin() {
 			}
 		}
 
+		if (min == Integer.MAX_VALUE) { // in case all channels are local
```
Comment (greghogan) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGateMetrics.java]: Is it worth removing the earlier check for `channels.isEmpty()` and merging the comments?

Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGateMetrics.java]: yes.

**Comment by zentol** [comment]
updated.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**